### ISSUE TITLE
Encryption support for "Non US Users"

### ISSUE THREAD

**Issue by 0x27** [brannondorsey/chattervox#23]
From the readme: "In the United States, it's illegal to broadcast encrypted messages on amateur radio frequencies.".

Would it be possible to perhaps add a config option to enable encryption support, for users in countries with less restrictive laws on amateur radio use? 

Or perhaps could you provide some pointers as to how this could be implemented? I guess it could be put in the same bit as the message signing / verification, but still getting familiar with the codebase...

**Comment by brannondorsey** [comment]
You are absolutely right that it _could_ be done by adding a bit in the flags byte in the header, similar to how compression is handled. That bit could indicate that the payload is encrypted. The question is, encrypted using _what_? A symmetric AES cypher is much more efficient than asymmetric pub/priv key encryption. But a _feature_ of Chattervox is that its connectionless, for better or worse, which can make key exchange tricky.

I'm hesitant to add this functionality as a first-class citizen of the protocol when it's so easy to add encryption _on top_ of the protocol. What I mean by that is, there is nothing stopping you from encrypting the payload contents of the chattervox packets.

Something like:

```bash
echo "ieTu7yeeTik2ee2fui8Zei4Eeo1MaiT3" > shared_secret.txt
echo "something secret, keep it safe" | aespipe -P shared_secret.txt | chattervox send
```

**Comment by skillet256** [comment]
Extending your thoughts on encrypting the payload above, what is a good example of how one would decrypt it with chattervox, assuming we know the shared secret?

**Comment by CowboyPilot** [comment]
Easy way to accomplish this is with a simple AES256 app on whatever computing device you're using.  Play Story has an app literally called "AES 256".  You input the shared key, type in the window, copy the output send and then copy what you recevie, paste into the window to decrypt.  yes it's an extra step but it will work with literally anything without having to have internal program support (Ribbit, Rattlegram, APRS, Chattervox, JS8, FSK, etc)
